dotenv/parse.ts

```diff
@@ -10,7 +10,9 @@ type LineParseResult = {
 type CharactersMap = { [key: string]: string };
 
 const RE_KEY_VALUE =
-  /^\s*(?:export\s+)?(?<key>[a-zA-Z_]+[a-zA-Z0-9_]*?)\s*=[\ \t]*('\n?(?<notInterpolated>(.|\n)*?)\n?'|"\n?(?<interpolated>(.|\n)*?)\n?"|(?<unquoted>[^\n#]*)) *#*.*$/gm;
+  /^\s*(?:export\s+)?(?<key>[^\s=#]+?)\s*=[\ \t]*('\n?(?<notInterpolated>(.|\n)*?)\n?'|"\n?(?<interpolated>(.|\n)*?)\n?"|(?<unquoted>[^\n#]*)) *#*.*$/gm;
```

Comment:
Can you please explain this regex change?

Reply:
The key part `(?<key>[a-zA-Z_]+[a-zA-Z0-9_]*?)` has been changed to `(?<key>[^\s=#]+?)`.

Now this regexp matches even with invalid key entries (like `1A=FOO`), and the key validation is performed later inside the while loop. That makes us possible to warn at invalid keys. Previously the lines with invalid keys were completely skipped and there was no room to warn with it.